packages/core/src/auth/auth.ts

```diff
@@ -842,16 +842,25 @@ export class Auth implements AuthService, ConnectionManager {
      * Auth processes can fail due to reasons outside of authentication actually being expired.
      * We do not want to intepret these failures as invalid/expired auth tokens.
      *
-     * We use this to check if the given error is unanticipated. If it is we throw,
-     * expecting the caller to not change the state of the connection.
+     * We use this to check if the given error is recoverable, meaning retrying getToken at
+     * a later time could succeed. If it is recoverable, we throw, expecting the caller to not
+     * change the state of the connection.
      */
-    private throwOnNetworkError(e: unknown) {
+    private throwOnRecoverableError(e: unknown) {
         if (isNetworkError(e)) {
             throw new ToolkitError('Failed to update connection due to networking issues', {
                 cause: e,
                 code: e.code,
             })
         }
+
+        const possibleCacheError = DiskCacheError.instanceIf(e)
+        if (possibleCacheError instanceof DiskCacheError) {
```

Comment:
can we use the [`is` pattern](https://github.com/aws/aws-toolkit-vscode/blob/4c499f5501fff761d45ca17da78084ddafa27233/packages/core/src/shared/errors.ts#L558) instead of this `instanceIf` thing.

oh, `instanceIf` is also a "factory" I guess. we need to find a more conventional pattern. 

```suggestion
        if (isDiskCacheError(e)) {
```

Reply:
Yeah this creates a new instance instead of verifying it meets a condition.

The reason for why used this over an "is" is in this comment: https://github.com/aws/aws-toolkit-vscode/pull/5549#discussion_r1746199261